File: Machine_Learning/Regression/Simple_Linear_Regression/template.R
#Simple Linear Regression

#Importing Dataset
dataset=read.csv('Salary_Data.csv')
#dataset=dataset[,2:3]

#Splitting dataset into training and test set
#install.packages('caTools')
#Click the checkbox of caTools OR 
library(caTools)
set.seed(123)
split=sample.split(dataset$Salary, SplitRatio=2/3, )
training_set=subset(dataset,split==TRUE)
test_set=subset(dataset, split==FALSE)

#Feature Scaling
#training_set[,2:3]=scale(training_set[, 2:3])
#test_set[, 2:3]=scale(test_set[, 2:3])

#Fitting Simple Linear Regression to the training set
regressor=lm(formula=Salary~YearsExperience, data=training_set)

#Prediciting the test set results
y_pred=predict(regressor, newdata=test_set)

#Visualising the training set results
install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary), 
             colour='red')+
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata=training_set)),
                colour='blue')+
  ggtitle('Salary vs Experience(Training set)')+
  xlab('Years of experience')+
  ylab('Salary')

#Visualising Test set results
install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary), 
             colour='red')+
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata=training_set)),
            colour='blue')+
  ggtitle('Salary vs Experience(Test set)')+
  xlab('Years of experience')+
  ylab('Salary')

File: Machine_Learning/Regression/Simple_Linear_Regression/template.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing dataset
dataset=pd.read_csv('Salary_Data.csv')
X=dataset.iloc[:, :-1].values
y=dataset.iloc[:,1].values

#Splitting dataset into train and test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train,y_test= train_test_split(X,y, test_size=1/3, random_state=0) 

"""#Feature scaling
from sklearn.preprocessing import StandardScaler
sc_X=StandardScaler()
X_train=sc_X.fit_transform(X_train)
X_test=sc_X.transform(X_test)"""

#Fitting simple linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor=LinearRegression()
regressor.fit(X_train, y_train)
